cups/snmp.c

```diff
@@ -467,7 +467,7 @@ _cupsSNMPStringToOID(const char *src,	/* I - OID string */
   * Range check input...
   */
 
-  if (!src || !dst || dstsize < 2)
+  if (!src || !dst || dstsize < 2 || !strchr(src, '.'))
```

Comment:
This precludes an OID consisting of a single number, which is valid for things like SNMP walk...